File: src/main/java/com/bdev/hogwarts_api/data/converter/cabinet/CabinetDtoConverter.java
package com.bdev.hogwarts_api.data.converter.cabinet;

import com.bdev.hogwarts_api.data.dto.cabinet.Cabinet;
import com.bdev.hogwarts_api.data.model.cabinet.CabinetModel;
import com.bdev.hogwarts_api.utils.EncodingUtils;

import static com.bdev.hogwarts_api.utils.EncodingUtils.toBase64;

public class CabinetDtoConverter {
    public static CabinetModel convert(Cabinet cabinet) {
        CabinetModel cabinetModel = new CabinetModel();

        cabinetModel.setId(cabinet.getId());
        cabinetModel.setName(toBase64(cabinet.getName()));
        cabinetModel.setCabinetType(cabinet.getCabinetType());

        return cabinetModel;
    }
}

File: src/main/java/com/bdev/hogwarts_api/data/converter/cabinet/CabinetDtoConverter.kt
package com.bdev.hogwarts_api.data.converter.cabinet

import com.bdev.hogwarts_api.data.dto.cabinet.ExistingCabinet
import com.bdev.hogwarts_api.data.dto.cabinet.NewCabinet
import com.bdev.hogwarts_api.data.model.cabinet.CabinetModel

import com.bdev.hogwarts_api.utils.EncodingUtils.toBase64

object CabinetDtoConverter {
    fun convertNew(cabinet: NewCabinet): CabinetModel {
        return CabinetModel(
                id = null,
                name = toBase64(cabinet.info.name)
        )
    }

    fun convertExisting(cabinet: ExistingCabinet): CabinetModel {
        return CabinetModel(
                id = cabinet.id,
                name = toBase64(cabinet.info.name)
        )
    }
}
